inst/stan/epinowcast.stan

```diff
@@ -103,20 +103,20 @@ transformed parameters{
       refp_sd = exp(refp_sd);
     }
     }
-    // calculate pmfs
-    profile("transformed_delay_reference_date_pmfs") {
-    for (i in 1:refp_fnrow) {
-      pmfs[, i] =
-         discretised_reporting_prob(refp_mean[i], refp_sd[i], dmax, model_refp,
-         2);
-    }
-    if (ref_as_p == 0) {
+    // calculate reference date logit hazards (unless no reporting effects)
+    profile("transformed_delay_reference_date_hazards") {
+    if (ref_as_p) {
       for (i in 1:refp_fnrow) {
-        ref_lh[, i] = prob_to_hazard(pmfs[, i]);
-        ref_lh[, i] = logit(ref_lh[, i]);
+        ref_lh[, i] =
```

Comment:
Having this as an option adds a fair bit of model complexity. I like it for the speed up it gives but could decide it is too much technical debt to carry around.

Reply:
I think you solved it elegantly in #140